cookies/views/resource.py

```diff
@@ -734,11 +736,21 @@ def resource_content(request, resource_id):
     else:
         content_type = 'application/octet-stream'
     if resource.file:
-        try:
-            with open(resource.file.path, 'rb') as f:
-                return HttpResponse(f.read(), content_type=content_type)
-        except IOError:    # Whoops....
-            return HttpResponse('Hmmm....something went wrong.')
+        if not settings.DEVELOP and (content_type in ('application/octet-stream', 'application/zip')):
+            # Serve specific content_type files through Nginx only if the
+            # Django app is not running using 'manage.py runserver'.
+            file_path = resource.file.path
+            file_name = os.path.basename(file_path)
+            response = HttpResponse(content_type=content_type)
+            response['Content-Disposition'] = 'attachment; filename=%s' % (file_name)
+            response['X-Accel-Redirect'] = posixpath.join(settings.MEDIA_URL, smart_str(file_path).lstrip(settings.MEDIA_ROOT))
+            return response
+        else:
+            try:
+                with open(resource.file.path, 'rb') as f:
+                    return HttpResponse(f.read(), content_type=content_type)
+            except IOError:    # Whoops....
```

Comment:
can we print the error to the log here please?

Reply:
Handled in the latest commit.